cloud/cloud.js

```diff
@@ -889,3 +892,203 @@ Moralis.Cloud.define(
+    return await checkRequesterCreatedTask(ethAddress, request.params.taskId)
+  },
+  {
+    fields: ["taskId"],
+    requireUser: true,
+  }
+);
+
+Moralis.Cloud.define(
+  "getTaskResponses",
+  async (request) => {
+    async function getResponses(taskId) {
+      const pipeline = [
+        { group: { objectId: "$questionId", data: { $push: "$$ROOT" }}},
+      ];
+      tableName = "Responses";
+      const Responses = Moralis.Object.extend(tableName);
+      const query = new Moralis.Query(Responses);
+      const res = await query.equalTo("taskId", taskId).aggregate(pipeline);
+      return res;
+    }
+    const taskId = request.params.taskId;
+    const ethAddress = request.user.get("ethAddress");
+    if (!(await checkRequesterCreatedTask(ethAddress, taskId))) return [];
+    const responses = getResponses(taskId);
+    return responses;
+  },
+  {
+    fields: ["taskId"],
+    requireUser: true,
+  }
+);
+
+Moralis.Cloud.define(
+  "getTaskUsers",
+  async (request) => {
+    async function getTaskUsers(taskId) {
+      tableName = "TaskUsers";
+      const TaskUsers = Moralis.Object.extend(tableName);
+      const query = new Moralis.Query(TaskUsers);
+      query.equalTo("taskId", taskId);
+      query.notEqualTo("status", 0);
+      const res = await query.find();
+      return res;
+    }
+    const taskId = request.params.taskId;
+    const ethAddress = request.user.get("ethAddress");
+    if (!(await checkRequesterCreatedTask(ethAddress, taskId))) return [];
+    const taskUsers = getTaskUsers(taskId);
+    return taskUsers;
+  },
+  {
+    fields: ["taskId"],
+    requireUser: true,
+  }
+);
+
+Moralis.Cloud.define(
+  "getUserResponse",
+  async (request) => {
+    async function getTaskUsers(taskId, taskerId) {
+      tableName = "Responses";
+      const Responses = Moralis.Object.extend(tableName);
+      const query = new Moralis.Query(Responses);
+      query.equalTo("taskId", taskId);
+      query.equalTo("userId", taskerId);
+      const res = await query.find();
+      return res;
+    }
+    const taskId = request.params.taskId;
+    const ethAddress = request.user.get("ethAddress");
+    if (!(await checkRequesterCreatedTask(ethAddress, taskId))) return [];
+    const userResponse = getTaskUsers(taskId, request.params.taskerId);
+    return userResponse;
+  },
+  {
+    fields: ["taskId", "taskerId"],
+    requireUser: true,
+  }
+);
+
+Moralis.Cloud.define(
+  "updateApprovalStatus",
+  async (request) => {
+    const taskId = request.params.taskId;
+    const ethAddress = request.user.get("ethAddress");
+    if (!(await checkRequesterCreatedTask(ethAddress, taskId)))  {
+      return {
+        success: false,
+        message: "Approval update failed - unauthorized update!"
+      };
+    }
+    tableName = "TaskUsers";
+    const TaskUsers = Moralis.Object.extend(tableName);
+    const query = new Moralis.Query(TaskUsers);
+    query.get(request.params.objectId).then(
+      (taskUser) => {
```

Comment:
Here, can you first check that `taskUser.get("status")` is `1` (i.e. not reviewed), and if it's not, then return an error? I think that this would ensure correct behavior: the status can only be changed from `1` to something else.

Reply:
done